File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        
        length = len(word1) if len(word1)<len(word2) else len(word2)
        word = ''
        for i in range(length):
            word += word1[i] + word2[i]

        if len(word1)<len(word2):
            word+=word2[length::]
        elif len(word1)>len(word2):
            word+=word1[length::]
        print(word)
        return word

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        

        int i=0;
        int j=0;
        StringBuilder res = new StringBuilder();
        while(i<word1.length() && j<word2.length()){
            res.append(word1.charAt(i));
            res.append(word2.charAt(j));
            i++;
            j++;
        }
        System.out.println(i +" "+ j);
        if(word1.length()>i){
            while(i<word1.length()){
                res.append(word1.charAt(i));
                i++;
            }
        }
        else{
            while(j<word2.length()){
                res.append(word2.charAt(j));
                j++;
            }
        }
        return res.toString();
    }
}